File: ruby/exercises/guess-the-number.rb
require 'colorize'
require_relative('../helpers/exitPrompt')
require_relative('../helpers/quitToMenu')

class GuessTheNumber

  def initialize
    @computerNumber = generateNumber
  end

  def generateNumber
    Random.rand(100)
  end

  def promptForInput
    @input = gets.sub(/\n/, '')
    quitToMenu(@input)

    if (@input.to_i == @computerNumber)
      puts("You got the right number!".yellow)
      exitPrompt
    else
      puts("Guess again!".green)
      promptForInput
    end
  end

  def start!
    puts("Guess the number I'm thinking of!".green)
    generateNumber
    promptForInput
  end
end

GuessTheNumber.new.start!




# -----



class GuessTheNumber
end

def generateNumber
  Random.rand(100)
end

def playGuessTheNumber
  number = generateNumber
  guess = nil

  while guess !== number
    puts("Guess the number I'm thinking of!".green)
    guess = promptForGuess

      puts("You got the right number!".yellow)
      exitPrompt
    else
      puts("Guess again!".green)
      promptForInput
    end
  begin
  end while guess !== number
end

def promptForGuess
  computerNumber = generateNumber
  @input = gets.sub(/\n/, '')
  quitToMenu(@input)
  @input.to_i

end

playGuessTheNumber

GuessTheNumber.new

File: node/exercises/guess-the-number.js
const colors = require('colors')
const quitCommand = require('../helpers/quitCommand')
let computerNumber;

function generateNumber(){
  computerNumber = Math.ceil(Math.random() * 100)
}

generateNumber();
console.log("Guess the number I'm thinking of!".green)

process.stdin.resume();
process.stdin.setEncoding('utf8');

process.stdin.on('data', function(input){
  if(input.indexOf('/') === 0) {
    quitCommand(input, process)
  } else {
    input = Number(input.replace(/\D/, ''))

    if(input === computerNumber){
      console.log('Congratulations! You guessed the right number!'.green)
      process.disconnect();
    } else if (input === NaN){
      console.log('Guesses must be a number!'.green)
    } else {
      console.log('Guess again!'.green)
    }
  }
})
